### PR TITLE
[FLINK-33471] Make flink kubernetes operator compilable with jdk21

### PR DESCRIPTION
**Pull request by snuyanzin**
## What is the purpose of the change

The PR is aiming to support compilation with java 21

Since spotless and google java format needs to be updated to work with java 21
new spotless/google-java-format rules are applied as well

Also lombok dependency is bump to the version which supports  java 21

## Verifying this change
This change is a trivial rework

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes )
  - The public API, i.e., is any changes to the `CustomResourceDescriptors`: (no)
  - Core observer or reconciler logic that is regularly executed: (no)

## Documentation

  - Does this pull request introduce a new feature? ( no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Comment by snuyanzin** [comment]
@1996fanrui @gyfora since you've looked at similar PR for java 17 support may I ask you to have a look at java 21 support?

File: pom.xml [hunk 1]
```diff
@@ -259,7 +259,7 @@ under the License.
                 <configuration>
                     <java>
                         <googleJavaFormat>
-                            <version>1.7</version>
+                            <version>1.17.0</version>
```
Comment (snuyanzin) [inline pom.xml:262]: minimum version supporting java 21

**Review by 1996fanrui — APPROVED** [review]
Thanks @snuyanzin for the contribution!

LGTM.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -26,7 +26,7 @@ jobs:
     name: test_ci
     strategy:
       matrix:
-        java-version: [ 11, 17 ]
+        java-version: [ 11, 17, 21 ]
```
Comment (mxm) [inline .github/workflows/ci.yml:29]: Is it worth to explode the test matrix by adding another Java version? Can't we test Java 21 compatibility without running all integration tests with it?

Comment (snuyanzin) [reply .github/workflows/ci.yml:29]: it does not run all
it runs only against flink1.18 (same as for 17)
below there are exclusions for 1.16 and 1.17

I'm ok to reduce number of tests, however then would be great if you can provide a hint what else could be cut

Comment (mxm) [reply .github/workflows/ci.yml:29]: Oh! I missed that you were excluding. This is fine then.

**Review by mxm — APPROVED** [review]

**Pull request merged**